src/chunk.js

```diff
@@ -73,15 +74,21 @@ function indexToPos (index, sectionY) {
   return new Vec3(x, sectionY * 16 + y, z)
 }
 
-function readBlocks (chunk, sectionY, blocks) {
+function readBlocks (chunk, mask, sectionY, blocks) {
   blocks = Buffer.from(blocks)
   for (let index = 0; index < blocks.length; index++) {
     const blockType = blocks.readUInt8(index)
     const pos = indexToPos(index, sectionY)
+    if(blockType) updateBitMap(mask, pos.y) // if not "empty"
     chunk.setBlockType(pos, blockType)
   }
 }
 
+function updateBitMap(mask, y) {
+  y = Math.floor(y / 16)
+  mask.bitMap |= 1 << y
+}
```

Comment:
**TL;DR** everything is here ^

![image](https://user-images.githubusercontent.com/11330271/50567449-9b42cd00-0d4d-11e9-9627-abb512666a65.png)